src/lib/DepositApiClient.js

```diff
@@ -97,12 +97,20 @@ export class RDMDepositApiClient extends DepositApiClient {
       );
       return new DepositApiClientResponse(data, errors);
     } catch (error) {
-      const data = this.recordSerializer.deserialize(error.response.data || {});
+      const responseData = error.response.data;
+      const data = this.recordSerializer.deserialize(responseData || {});
+
+      const noFieldErrors = !('errors' in responseData);
```

Comment:
minor: `prop in object` operator checks for `prop` in an object **and its prototype**. It is safer to look using `object.hasOwnProperty('prop')` if inheritance from prototype is not desired.